core/dbt/parser/manifest.py

```diff
@@ -599,6 +651,115 @@ def write_manifest_for_partial_parse(self):
         except Exception:
             raise
 
+    def build_public_nodes(self) -> bool:
+        """This method loads the dependencies from dependencies.yml, reads in the
+        the publication artifacts and adds the PublicModels to the manifest
+        "public_nodes" dictionary. It returns a boolean that indicates that
+        public nodes have been rebuilt."""
```

Comment:
This method is pretty long and involved - If there is a way to split it up into two - one for building public nodes (perhaps returning relevant saved artifacts), and one (private method) for detecting changes to public nodes - it would be helpful for readability. Could be a follow-on refactor

Reply:
I moved the straightforward loading of new public nodes into a separate method. May not help a lot... The rest is a bit harder to pull apart because of detecting changes.